File: backend/app/schemas/analytics.py
from datetime import datetime

from pydantic import BaseModel


class AnalyticsReportPoint(BaseModel):
    session_id: str
    target_role: str
    created_at: datetime
    overall_score: int
    logic_score: int
    technical_score: int
    expression_score: int
    project_depth_score: int


class AbilityAverages(BaseModel):
    logic_score: float
    technical_score: float
    expression_score: float
    project_depth_score: float


class WeakestDimension(BaseModel):
    key: str
    label: str
    average_score: float
    max_score: int


class AnalyticsImprovementPlanItem(BaseModel):
    priority: str
    topic: str


class AnalyticsOverview(BaseModel):
    completed_interview_count: int
    average_overall_score: float
    latest_report: AnalyticsReportPoint | None
    ability_averages: AbilityAverages | None
    weakest_dimension: WeakestDimension | None
    latest_improvement_plan: list[AnalyticsImprovementPlanItem]


class AnalyticsTrendItem(BaseModel):
    session_id: str
    target_role: str
    report_created_at: datetime
    overall_score: int
    logic_score: int
    technical_score: int
    expression_score: int
    project_depth_score: int


class AnalyticsTrendResponse(BaseModel):
    items: list[AnalyticsTrendItem]


class AnalyticsHistoryItem(AnalyticsTrendItem):
    difficulty: str
    interview_type: str
    knowledge_base_names: list[str]

File: frontend/src/types/analytics.ts
import type { PageResponse } from '../api/types'

export type AnalyticsReportPoint = {
  session_id: string
  target_role: string
  created_at: string
  overall_score: number
  logic_score: number
  technical_score: number
  expression_score: number
  project_depth_score: number
}

export type AbilityAverages = {
  logic_score: number
  technical_score: number
  expression_score: number
  project_depth_score: number
}

export type WeakestDimension = {
  key: keyof AbilityAverages
  label: string
  average_score: number
  max_score: number
}

export type AnalyticsImprovementPlanItem = {
  priority: string
  topic: string
}

export type AnalyticsOverview = {
  completed_interview_count: number
  average_overall_score: number
  latest_report: AnalyticsReportPoint | null
  ability_averages: AbilityAverages | null
  weakest_dimension: WeakestDimension | null
  latest_improvement_plan: AnalyticsImprovementPlanItem[]
}

export type AnalyticsTrendItem = {
  session_id: string
  target_role: string
  report_created_at: string
  overall_score: number
  logic_score: number
  technical_score: number
  expression_score: number
  project_depth_score: number
}

export type AnalyticsTrendResponse = {
  items: AnalyticsTrendItem[]
}

export type AnalyticsHistoryItem = AnalyticsTrendItem & {
  difficulty: string
  interview_type: string
  knowledge_base_names: string[]
}

export type AnalyticsHistoryResponse = PageResponse<AnalyticsHistoryItem>
